### PR TITLE
Add Set Slot and Get Slot VIs to the API.

### PR DESCRIPTION
**Pull request by Sarci1**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-scan-engine-ethercat-custom-device/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

This PR adds functions to set a slot to a module class and get a slot as a module class including supporting VIs to do so.

### Why should this Pull Request be merged?

Adds the additional functionality for getting and setting modules, i.e. scripting the module into the system definition and reading the system definition behavior into the appropriate class.

### What testing has been done?

Validated that the unit tests work.

### CODE REVIEW

**Comment by rtzoeller** [comment]
- [ ] I am seeing a large number of files which appear to be reverted compared to `main`. For example, `Custom Device Source/Scripting/Modules/Channel Classes/Counter Driven Output Channel/Read Counter Driven State.vi` no longer has source separated from compiled code. Can you check the VIs included in this PR to ensure they are correct?

File: Custom Device Source/Docs/scanengineveristand-errors.txt [hunk 1]
```diff
@@ -51,4 +51,10 @@ The Local Chassis already exists.
 <nierror code="537719">
 An insufficient number of channels was provided for module creation.
 </nierror>
+<nierror code="537720">
+The module could not be added to the slot because it was not found in the list of supported modules.
+</nierror>
+<nierror code="537721">
+This module and/or mode is not currently supported by the API.
+</nierror>
```
Comment (rtzoeller) [inline Custom Device Source/Docs/scanengineveristand-errors.txt:59]: Why do we need two error codes for this? It seems like 537721 covers it all.

**Review by rtzoeller — COMMENTED** [review]
- [ ] I'm seeing an issue with some VIs not being part of `Scan Engine Scripting.lvlib`, despite the library thinking it owns them. E.g. `Update Channels.vi`.

- [ ] In general, I'd prefer if VI names started with a verb (the same as for functions in a text based language). E.g. `Channel Names and Default Values from Slot.vi` could be called `Get Channel Names and Default Values for Slot.vi`. I'm sure we're inconsistent about this, but it'd be good to start moving in the right direction.

- [x] For symmetry, `Update Channels.vi` should probably be called `Set Channel Names and Default Values for Slot.vi`.

- [x] We're hardcoding the error strings in `Get/Set Slot.vi` - can we use the ones provided by `Error Strings.vi`, or otherwise single-source them in the scripting API? Same for `Module Model to Module Class.vi`.

- [x] The regex in `Set Slot.vi` doesn't quite match the one in the regex101 link (trailing `$`). Ironically @Karl-G1 called me out for doing this earlier today.

- [x] In `Set Slot.vi`, remove the "1" from the outer case structure (i.e. just make the case "Default"). Unless "1" is significant for some reason worth calling out that I'm missing.

- [ ] It seems like the inner case structure of `Set Slot.vi` could be a dynamic dispatch on the `Mode` object? We're using the class name to choose which code to run, which screams dynamic dispatch to me... on the other hand, `Get Slot.vi` is using a `DIO Mode` numeric to choose which code to run... so maybe we should stay way from dynamic dispatch for symmetry? No action required yet, I need to think about this.

**Comment by Sarci1** [comment]
> * [x]  I'm seeing an issue with some VIs not being part of `Scan Engine Scripting.lvlib`, despite the library thinking it owns them. E.g. `Update Channels.vi`.
> * [x]  In general, I'd prefer if VI names started with a verb (the same as for functions in a text based language). E.g. `Channel Names and Default Values from Slot.vi` could be called `Get Channel Names and Default Values for Slot.vi`. I'm sure we're inconsistent about this, but it'd be good to start moving in the right direction.
> * [x]  For symmetry, `Update Channels.vi` should probably be called `Set Channel Names and Default Values for Slot.vi`.
> * [x]  We're hardcoding the error strings in `Get/Set Slot.vi` - can we use the ones provided by `Error Strings.vi`, or otherwise single-source them in the scripting API? Same for `Module Model to Module Class.vi`.
> * [x]  The regex in `Set Slot.vi` doesn't quite match the one in the regex101 link (trailing `$`). Ironically @Karl-G1 called me out for doing this earlier today.
> * [x]  In `Set Slot.vi`, remove the "1" from the outer case structure (i.e. just make the case "Default"). Unless "1" is significant for some reason worth calling out that I'm missing.
> * [ ]  It seems like the inner case structure of `Set Slot.vi` could be a dynamic dispatch on the `Mode` object? We're using the class name to choose which code to run, which screams dynamic dispatch to me... on the other hand, `Get Slot.vi` is using a `DIO Mode` numeric to choose which code to run... so maybe we should stay way from dynamic dispatch for symmetry? No action required yet, I need to think about this.
Yes, we definitely can dynamic dispatch here. The reason I didn't was because I wanted these classes to be data constructs for the modules and not manipulate the system definition. Putting it in a dynamic dispatch would change that. Think on it though, because I can change it.

**Comment by rtzoeller** [comment]
- [ ] `Get/Set Channel Names and Default Values from Slot.vi` aren't in the same place on disk, but should be. They probably should use "for" instead of "from" in the names (IMO it reads better).
- [ ] It looks like `Channel Names and Default Values from Slot.vi` and `Update Channels.vi` are still on disk, and can be removed.

> Yes, we definitely can dynamic dispatch here. The reason I didn't was because I wanted these classes to be data constructs for the modules and not manipulate the system definition. Putting it in a dynamic dispatch would change that. Think on it though, because I can change it.

I like that argument. Consider this point dropped, unless @buckd disagrees.

- [ ] Consider adding a boolean output to `Get Slot.vi` to return "Is Empty?". That would save clients from needing to inspect the stored model directly.

**Review by rtzoeller — APPROVED** [review]

**Comment by Sarci1** [comment]
> * [x]  `Get/Set Channel Names and Default Values from Slot.vi` aren't in the same place on disk, but should be. They probably should use "for" instead of "from" in the names (IMO it reads better).
> * [x]  It looks like `Channel Names and Default Values from Slot.vi` and `Update Channels.vi` are still on disk, and can be removed.
> 
> > Yes, we definitely can dynamic dispatch here. The reason I didn't was because I wanted these classes to be data constructs for the modules and not manipulate the system definition. Putting it in a dynamic dispatch would change that. Think on it though, because I can change it.
> 
> I like that argument. Consider this point dropped, unless @buckd disagrees.
> 
> * [x]  Consider adding a boolean output to `Get Slot.vi` to return "Is Empty?". That would save clients from needing to inspect the stored model directly.

I used "empty" instead of "Is Empty?" to stay aligned with a similar construct used elsewhere: "found" instead of "Is Found?"

**Review by buckd — APPROVED** [review]

**Pull request merged**